src/components/QueryHistory.js

```diff
@@ -121,4 +123,22 @@ export class QueryHistory extends React.Component {
     const queries = historyQueries.concat(favoriteQueries);
     this.setState({ queries });
   };
+
+  editLabel = (query, variables, operationName, label, favorite) => {
+    const item = {
+      query,
+      variables,
+      operationName,
+      label,
+    };
+    if (favorite) {
+      this.favoriteStore.edit({ ...item, favorite });
+    } else {
+      this.historyStore.edit(item);
+    }
+    const historyQueries = this.historyStore.items;
+    const favoriteQueries = this.favoriteStore.items;
+    const queries = historyQueries.concat(favoriteQueries);
```

Comment:
nit: could just be:
```
this.setState({ ...this.historyStore.items, ...this.favoriteStore.items });
```

Reply:
Ooo brilliant, I just copied the code in `toggleFavorite`, but refactoring to save unused variables is a great call. I'll edit both methods.